syntax = "proto3";

package processeddata;

message ProcessedData {
    uint32 sequence_num = 1;
    repeated OutputDataSet outputdataset = 2; // 1 set for every frequency
    double processing_time = 3;
    double initialization_time = 4;
    double sequence_start_time = 5;
    string rf_samples_location = 6;

    message OutputDataSet {
        repeated ComplexData mainacf = 1;
        repeated ComplexData intacf = 2;
        repeated ComplexData xcf = 3;
        uint32 numberofbeams = 4;
        uint32 numberofranges = 5;
        uint32 numberoflags = 6;
        repeated BeamData beamformedsamples = 7;
        repeated DebugData debugsamples = 8;
        uint32 slice_id = 9;


        message BeamData {
            uint32 beamnum = 1;
            repeated ComplexData mainsamples = 2;
            repeated ComplexData intfsamples = 3;
        }

        message DebugData {
            string stagename = 1;
            repeated AntennaData antennadata = 2;
            message AntennaData {
                repeated ComplexData antennasamples = 1;
            }
        }

        message ComplexData {
            float real = 1;
            float imag = 2;
        }
    }
}
